### PR TITLE
UOE-11349: Update Lineitemtype

### PR DESCRIPTION
**Pull request by pm-isha-bharti**

### CODE REVIEW

**Review by Pubmatic-Dhruv-Sonone — APPROVED** [review]

File: tasks/update.py [hunk 1]
```diff
@@ -249,10 +239,155 @@ def update(self):
...
+        if self.setting_class.NEW_LINEITEM_TYPE not in ("NETWORK", "HOUSE", "PRICE_PRIORITY", "SPONSORSHIP"):
+            self.logger.info('NEW_LINEITEM_TYPE supports only one of the value ("NETWORK", "HOUSE", "PRICE_PRIORITY", "SPONSORSHIP")')
+            return False
+        return True
+
+    def get_line_items(self):
+        return UpdateUtils.get_line_items(self)
+
+    def print_skipped_line_items(self, skip_line_items):
+        UpdateUtils.print_skipped_line_items(skip_line_items)
```
Comment (Pubmatic-Dhruv-Sonone) [inline tasks/update.py:288]: Do we need this function?

File: tasks/update.py [hunk 2]
```diff
@@ -249,10 +239,155 @@ def update(self):
...
+            line_item_service = self.ad_manager_client.GetService('LineItemService', version=self.API_VERSION)
+            returned_line_items=line_item_service.updateLineItems(line_items_to_update)
+            if len(returned_line_items) > 0:
+                self.logger.info("\nSuccessfully updated following line items")
+                for line_item in returned_line_items:
+                    self.logger.info(f"{self.color.BLUE}{line_item['name']}{self.color.END}")
+            else:
+                self.logger.info("No line items were updated.")
+        else:
+            self.logger.info("No line items matched the criteria for update.")
```
Comment (Pubmatic-Dhruv-Sonone) [inline tasks/update.py:385]: Can we make this a separate function?  `VideoPositionUpdater` also have similar update code.

**Review by Pubmatic-Dhruv-Sonone — CHANGES_REQUESTED** [review]

File: tasks/update.py [hunk 3]
```diff
@@ -249,10 +239,155 @@ def update(self):
...
+                    line_item['primaryGoal']['goalType'] = 'DAILY'
+                    line_item['primaryGoal']['units'] = 100
+                line_item['skipInventoryCheck'] = True
+                line_item['allowOverbook'] = True
+                line_item['priority'] = 4
+
+            elif self.setting_class.NEW_LINEITEM_TYPE == "PRICE_PRIORITY":
+                line_item['priority'] = 12
+
+            line_items_with_current_type[line_item['name']]=current_type
```
Comment (Pubmatic-Dhruv-Sonone) [inline tasks/update.py:349]: Should this be `lineItemType` instead of 'name'?

File: tasks/update.py [hunk 4]
```diff
@@ -249,10 +239,155 @@ def update(self):
...
+        If the update is successful, the names of the updated line items are logged.
+        """
+        response = self.get_line_items()
+        line_items = response['results'] if 'results' in response else []
+        if len(line_items ) <= 0:
+            self.logger.info("No line item found for given input")
+            return
+
+        line_items_to_update, line_items_with_current_type = self.select_line_items_to_update(line_items)
+        self.print_line_items_with_current_type(line_items_with_current_type)
```
Comment (Pubmatic-Dhruv-Sonone) [inline tasks/update.py:368]: Why do we need to print this?

File: tasks/update.py (hunk 1 above)
Comment (pm-isha-bharti) [reply tasks/update.py:288]: Not required, will remove

**Review by Pubmatic-Dhruv-Sonone — APPROVED** [review]

**Pull request merged**